File: Chapter 04/4.2 Downward Dependency/Before/PresentationLayer.cs
namespace Before
{
    // A fake UI layer class to illustrate the bad dependency
    public class PresentationLayer
    {
        public static PresentationLayer Instance { get; } = new PresentationLayer();
        public void UpdateStatusLabel(string text)
        {
            Console.WriteLine($"[UI UPDATE]: {text}");
        }
    }
}

File: Chapter 04/Python/section_4_2_downward_dependency/before/presentation_layer.py
"""
A fake UI layer class to illustrate the bad dependency.
"""
from shared.log_manager import LogManager
class PresentationLayer:
    _instance = None

    @classmethod
    def get_instance(cls):
        if cls._instance is None:
            cls._instance = PresentationLayer()
        return cls._instance

    def update_status_label(self, text: str):
        LogManager.info("PresentationLayer", "[UI UPDATE]: {0}", text)